qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -588,13 +597,12 @@ def closeEvent(self, event):
             import matplotlib.pyplot as plt  # noqa
             plt.close('all')
 
+            # Close any remaining windows
+            app.closeAllWindows()
```

Comment:
There was a `if app is not None` condition before. I do not know if this is really necessary but you used it above (line 582). Maybe you could keep it here too.

Reply:
Good catch, I'd missed that